ports/servoshell/egl/ohos.rs

```diff
@@ -185,6 +203,56 @@ impl ServoAction {
                 servo.present_if_needed();
             },
             Resize { width, height } => servo.resize(Coordinates::new(0, 0, *width, *height)),
+            FocusWebview(id) => {
+                if let Some(native_webview_components) =
+                    NATIVE_WEBVIEWS.lock().unwrap().get(*id as usize)
+                {
+                    if (servo.active_webview().id() != native_webview_components.id) {
+                        servo.focus_webview(native_webview_components.id);
+                        servo.pause_compositor();
+                        let (window_handle, _, coordinates) = simpleservo::get_raw_window_handle(
+                            native_webview_components.xcomponent.0,
+                            native_webview_components.window.0,
+                        );
+                        servo.resume_compositor(window_handle, coordinates);
+                        let url = servo
+                            .active_webview()
+                            .url()
+                            .map(|u| u.to_string())
+                            .unwrap_or(String::from("about:blank"));
+                        SET_URL_BAR_CB
+                            .get()
+                            .map(|f| f.call(url, ThreadsafeFunctionCallMode::Blocking));
+                    }
+                } else {
+                    error!("Could not find webview to focus");
+                }
+            },
+            NewWebview(xcomponent, window) => {
+                servo.pause_compositor();
+                servo.new_toplevel_webview("about:blank".parse().unwrap());
+                let (window_handle, _, coordinates) =
+                    simpleservo::get_raw_window_handle(xcomponent.0, window.0);
+
+                servo.resume_compositor(window_handle, coordinates);
+                let webview = servo.newest_webview().expect("There should always be one");
+                let id = webview.id();
+                NATIVE_WEBVIEWS
+                    .lock()
+                    .unwrap()
+                    .push(NativeWebViewComponents {
+                        id: id,
+                        xcomponent: xcomponent.clone(),
+                        window: window.clone(),
+                    });
+                let url = webview
+                    .url()
+                    .map(|u| u.to_string())
+                    .unwrap_or(String::from("about:blank"));
+                SET_URL_BAR_CB
```

Comment:
I think I would modify the callback to have both the webview id (From servo) and the URL as parameters. 
On the ArkTS side, you could keep a map of servo_webview_id -> url.